internal/services/v1/permissions_test.go

```diff
@@ -1403,3 +1405,117 @@ func TestGetCaveatContext(t *testing.T) {
 	require.NoError(t, err)
 	require.Contains(t, caveatMap, "foo")
 }
+
+func TestLookupResourcesWithCursors(t *testing.T) {
```

Comment:
It might be tricky but I think a nice test to have is to show that for a long list of results, we start to stream as soon as possible, that it does not wait for the whole response (as we did before) and that it does not load the whole thing, but only the relationships needed.. E.g. we could somehow test that there is only a single chunked dispatch for a very wide relation, or that certain relationships are never loaded by the datastore (e.g. with a wrapper around the datastore)

Reply:
Any ideas on how to tell that from the caller's perspective? From here, a "waiting internally" would just be a larger delay before time of first result.